File: backend/models/message.go
package models

import "time"

// messages belong to conversation with have two keys
// being the respective user ids
type Message struct {
	ConversationID string `json:"conversation_id"`
	From           string `json:"from"`
	To             string `json:"to"`
	Message        string `json:"message"`
	CreatedAt      string `json:"created_at"`
	Delivered      bool   `json:"delivered"`
	Read           bool   `json:"read"`
}

func (m *Message) NewMessage(convID string, from string, to string, message string) *Message {
	return &Message{
		ConversationID: convID,
		From:           from,
		To:             to,
		Message:        message,
		CreatedAt:      time.Now().String(),
		Delivered:      false,
		Read:           false,
	}
}

File: python/models/message.py
from pydantic import BaseModel
from datetime import datetime


class Message(BaseModel):
    conversation_id: str
    from_: str
    to: str
    message: str
    created_at: str = datetime.now().isoformat()
    delivered: bool = False
    read: bool = False

    @classmethod
    def new_message(cls, conv_id: str, from_: str, to: str, message: str):
        return cls(
            conversation_id=conv_id,
            from_=from_,
            to=to,
            message=message,
            created_at=datetime.now().isoformat(),
            delivered=False,
            read=False
        )
    
    